photutils/detection/findstars.py

```
@@ -376,6 +377,135 @@ def find_stars(self, data):
+
+    wcs : `~astropy.wcs.WCS`
+        The WCS transformation to use to convert from pixel coordinates
+        to ICRS world coordinates.  If `None`, then the world
+        coordinates will not be returned in the output
+        `~astropy.table.Table`.
+    """
+
+    def __init__(self, threshold, box_size=3, footprint=None, mask=None,
+                 border_width=None, npeaks=np.inf, subpixel=True, error=None,
+                 wcs=None):
+        self.threshold = threshold
+        self.box_size = box_size
+        self.footprint = footprint
+        self.mask = mask
+        self.border_width = border_width
+        self.npeaks = npeaks
+        self.subpixel = subpixel
+        self.error = error
+        self.wcs = wcs
+
+    def __call__(self, data):
+        return self.find_stars(data)
+
+    def find_stars(self, data):
+        """
+        Parameters
+        ----------
+        data : array_like
+            The 2D array of the image.
+
+        Returns
+        -------
+        output : `~astropy.table.Table`
+            A table containing the x and y pixel location of the peaks and
+            their values.  If ``subpixel=True``, then the table will also
+            contain the local centroid and fitted peak value.
+        """
+
+        peaks = find_peaks(data, threshold=self.threshold,
+                           box_size=self.box_size, footprint=self.footprint,
+                           mask=self.mask, border_width=self.border_width,
+                           npeaks=self.npeaks, subpixel=self.subpixel,
+                           error=self.error, wcs=self.wcs)
+
+        if len(peaks) > 0:
+            if self.subpixel:
+                peaks['y_centroid'].name = 'ycentroid'
+                peaks['x_centroid'].name = 'xcentroid'
+            else:
+                peaks['y_peak'].name = 'ycentroid'
+                peaks['x_peak'].name = 'xcentroid'
+
+            peaks['id'] = np.arange(len(peaks)) + 1
+
```

Should raise an error here if sources were not found? @eteq